### PR TITLE
First version of asmrgen

### PR DESCRIPTION
**Pull request by alex4o**
The first version of the producer taht creates runtime code generators from llvm-tblgen exported json

### CODE REVIEW

**Comment by ltratt** [comment]
What is the copyright situation for the JSON files?

**Comment by alex4o** [comment]
They are licensed as:  Apache-2.0 WITH LLVM-exception, but I don't know how to check the copyright information?

**Comment by ltratt** [comment]
> They are licensed as: Apache-2.0 WITH LLVM-exception, but I don't know how to check the copyright information?

You're quite right, I was a bit sloppy with my language: I meant the licensing situation.

A vaguely related question is whether we need to even include them in this repo, or whether we can download them (presumably specific revisions) as part of a build file?

**Comment by alex4o** [comment]
They need to be generated and you need to download the whole llvm repository. Then use llvm-tblgen to generate it from there. I feel like that me too much effort, but I have the shell scripts to generate them in the repo.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,17 @@
 # cheri-instruction-rcg-gen
-Generate Runtime Code Generation library from llvm-tblgen json instruction spec
+Generate Runtime Code Generation library from json descriptions extracted from llvm-tblgen using `--dump-json`
+
+
+Build:
+
+```bash
```
Comment (ltratt) [inline README.md]: Can we make these just `sh` (they're not really bash specific, and some of us don't even use bash :) ).

Comment (alex4o) [reply README.md]: done

**Comment by ltratt** [comment]
> They need to be generated and you need to download the whole llvm repository. Then use llvm-tblgen to generate it from there. I feel like that me too much effort, but I have the shell scripts to generate them in the repo.

Ah. Is it important that we tie ourselves to a particular LLVM version (which hard-coding the JSON files) does or not? I guess "yes" but I'm not sure.

**Comment by alex4o** [comment]
Knowing that llvm is not stable in any way, I would like to be sure that this can be used after they have changed their definitions.

**Comment by ltratt** [comment]
Makes sense.

Please squash [what this means is: if you think some of the commits can usefully be squashed into one, please feel free to do so. You can a) use your discretion in which commits you squash b) you don't necessarily need to squash everything down to a single commit.].

**Comment by ltratt** [comment]
Did you mean to close this?

**Pull request closed without merging**